chain/network/src/routing_table_actor.rs

```diff
@@ -14,22 +16,355 @@ use crate::ibf::{Ibf, IbfBox};
     pub peer_ibf_set: IbfPeerSet,
+    /// Current view of the network. Nodes are Peers and edges are active connections.
+    pub raw_graph: Graph,
+    /// Active PeerId that are part of the shortest path to each PeerId.
+    pub peer_forwarding: Arc<HashMap<PeerId, Vec<PeerId>>>,
+    /// Last time a peer was reachable through with active edges.
+    pub peer_last_time_reachable: HashMap<PeerId, Instant>,
+    /// Everytime a group of peers becomes unreachable at the same time; We store edges belonging to
+    /// them in components. We remove all of those edges from memory, and save them do database,
+    /// If any of them were to be reachable again, we would re-add them.
+    ///
+    /// To store components, we have following column in the DB.
+    /// ColLastComponentNonce -> stores component_nonce: u64, which is the lowest nonce that
+    ///                          hasn't been used yet. If new component gets created it will use
+    ///                          this nonce.
+    /// ColComponentEdges     -> Mapping from `component_nonce` to list of edges
+    /// ColPeerComponent      -> Mapping from `peer_id` to last component nonce if there
+    ///                          exists one it belongs to.
+    store: Arc<Store>,
+    /// First component nonce id that hasn't been used. Used for creating new components.
+    pub next_available_component_nonce: u64,
+    /// True if edges were changed and we need routing table recalculation.
+    pub needs_routing_table_recalculation: bool,
+}
+
+impl RoutingTableActor {
+    pub fn new(my_peer_id: PeerId, store: Arc<Store>) -> Self {
+        let component_nonce = store
+            .get_ser::<u64>(ColLastComponentNonce, &[])
+            .unwrap_or(None)
+            .map_or(0, |nonce| nonce + 1);
+        Self {
+            edges_info: Default::default(),
+            #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+            peer_ibf_set: Default::default(),
+            raw_graph: Graph::new(my_peer_id),
+            peer_forwarding: Default::default(),
+            peer_last_time_reachable: Default::default(),
+            store,
+            next_available_component_nonce: component_nonce,
+            needs_routing_table_recalculation: false,
+        }
+    }
+
+    pub fn remove_edges(&mut self, edges: &[Edge]) {
+        for edge in edges.iter() {
+            self.remove_edge(edge);
+        }
+    }
+
+    pub fn remove_edge(&mut self, edge: &Edge) {
+        #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+        self.peer_ibf_set.remove_edge(&edge.to_simple_edge());
+
+        let key = (edge.peer0.clone(), edge.peer1.clone());
+        if self.edges_info.remove(&key).is_some() {
+            self.raw_graph.remove_edge(&edge.peer0, &edge.peer1);
+            self.needs_routing_table_recalculation = true;
+        }
+    }
+
+    /// `add_verified_edge` add edges that have already been verified.
```

Comment:
what does 'verified' mean?

Reply:
also mention that it returns true if edge was added succesfully.

you can  also remove the `add_verified_edge` from the beginning of the comment

Reply:
I meant that we checked their signatures. I updated comment to be 
```rust
    /// `add_verified_edge` adds edges, for which we already that theirs signatures
    /// are valid (`signature0`, `signature`).
```